src/test/common/clara/test_clear_ns_productions.cljc

```diff
@@ -0,0 +1,75 @@
+;;; Tests that clear-ns-productions! correction clears all vars marked as productions from the namespace.
+#?(:clj
+   (ns clara.test-clear-ns-productions
+     (:require [clara.tools.testing-utils :refer [def-rules-test
+                                                  side-effect-holder] :as tu]
+               [clara.rules :refer [fire-rules
+                                    insert
+                                    insert!
+                                    query
+                                    defrule
+                                    defquery
+                                    defsession
+                                    clear-ns-productions!]]
+               [clojure.test :refer [is deftest run-tests testing use-fixtures]]))
+   :cljs
+   (ns clara.test-clear-ns-productions
+     (:require [clara.rules :refer [fire-rules
+                                    insert
+                                    insert!
+                                    query]]
+               [cljs.test])
+     (:require-macros [clara.rules :refer [defrule defquery defsession clear-ns-productions!]]
+                      [cljs.test :refer [is deftest run-tests testing use-fixtures]])))
+
+(def before-clearing (atom nil))
+(defrule rule-to-be-cleared
+         [:a]
+         =>
+         (reset! before-clearing :before-clearing)
+         (insert! :before-clearing))
+
+(defquery query-to-be-cleared [] [?f <- :before-clearing])
```

Comment:
I should just delete it, because if you try to reference it after the call to `clear-ns-productions~`, you get a compiler error about an undefined symbol. I thought about adding a test to call `resolve` and validate that it doesn't exist anymore, which I'm happy to add if you think it's useful.

Reply:
I forgot to mention - you can query on the name of the query as well, which won't throw an exception.   See [here in the compiler](https://github.com/cerner/clara-rules/blob/0.16.1/src/main/clojure/clara/rules/compiler.clj#L1662) and the [query implementation in the engine](https://github.com/cerner/clara-rules/blob/0.16.1/src/main/clojure/clara/rules/engine.cljc#L1960).

Reply:
RIght, cool. Done.